ingest/src/test/java/com/microsoft/azure/kusto/ingest/E2ETest.java

```diff
@@ -429,6 +435,59 @@ private boolean canAuthenticate(ConnectionStringBuilder engineCsb) {
         return true;
     }
 
+    private void assertUrlCompare(String connectionDataSource, String clusterUrl, boolean autoCorrectEndpoint, boolean isQueued) {
+        if (!autoCorrectEndpoint || clusterUrl.contains(INGEST_PREFIX) || isReservedHostname(clusterUrl)) {
+            String host = clusterUrl.replaceFirst("https" + PROTOCOL_SUFFIX, "");
+            if (InetAddressUtils.isIPv6Address(host)) {
+                String compareString = clusterUrl.replaceFirst(PROTOCOL_SUFFIX, PROTOCOL_SUFFIX + '[') + ']';
+                assertEquals(compareString.toLowerCase(), connectionDataSource);
+            } else {
+                assertEquals(clusterUrl, connectionDataSource);
+            }
+        } else {
+            String compareString = isQueued ? clusterUrl.replaceFirst(PROTOCOL_SUFFIX, PROTOCOL_SUFFIX + INGEST_PREFIX) : clusterUrl;
+            assertEquals(compareString, connectionDataSource);
+        }
+    }
+
+    private static Stream<Arguments> provideStringsForAutoCorrectEndpointTests() {
+        return Stream.of(
+                Arguments.of(System.getenv("ENGINE_CONNECTION_STRING")),
+                Arguments.of("https://192.168.1.1"),
+                Arguments.of("https://2345:0425:2CA1:0000:0000:0567:5673:23b5"),
+                Arguments.of("https://127.0.0.1"),
+                Arguments.of("https://localhost"),
+                Arguments.of("https://onebox.dev.kusto.windows.net"));
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideStringsForAutoCorrectEndpointTests")
+    void testQueuedUseAutoCorrectEndpoint(String csb) {
+        ConnectionStringBuilder engineCsb = ConnectionStringBuilder.createWithAadApplicationCredentials(csb, appId, appKey, tenantId);
+        try {
+            QueuedIngestClient queuedIngestClient = IngestClientFactory.createClient(engineCsb, null, true);
+            assertUrlCompare(((QueuedIngestClientImpl) queuedIngestClient).connectionDataSource, engineCsb.getClusterUrl(), true, true);
+            queuedIngestClient = IngestClientFactory.createClient(engineCsb, null, false);
+            assertUrlCompare(((QueuedIngestClientImpl) queuedIngestClient).connectionDataSource, engineCsb.getClusterUrl(), false, true);
```

Comment:
Where are the tests for managedstreaming, which is the most complicated one?
Are you testing all the code you entered? all the new ctors?